electrumsv/nodeapi.py

```diff
@@ -711,6 +713,148 @@ async def jsonrpc_createrawtransaction_async(request: web.Request, request_id: R
     return Tx(1, transaction_inputs, transaction_outputs, locktime).to_hex()
 
 
+class TransactionInfo(TypedDict, total=False):
+    amount: float
+    blockhash: str | None
+    blockindex: int | None
+    blocktime: int | None
+    confirmations: int
+    details: list[TransactionDetails]
+    fee: float | None
+    generated: bool | None
+    hex: str
+    time: int
+    timereceived: int
+    trusted: bool | None
+    txid: str
+    walletconflicts: list[str]
+
+
+class TransactionDetails(TypedDict, total=False):
+    account: str
+    address: str | None
+    amount: float
+    category: str
+    fee: float | None
+    vout: int
+
+
+async def jsonrpc_gettransaction_async(request: web.Request, request_id: RequestIdType,
+        parameters: RequestParametersType) -> Any:
+    """
+    Get detailed information about a transaction in the wallet.
+    """
+    # Ensure the user is accessing either an explicit or implicit wallet.
+    wallet = get_wallet_from_request(request, request_id)
+    assert wallet is not None
+    # Similarly the user must only have one account (and we will ignore any
+    # automatically created petty cash accounts which we do not use yet).
+    accounts = wallet.get_visible_accounts()
+    if len(accounts) != 1:
+        raise web.HTTPInternalServerError(headers={ "Content-Type": "application/json" },
+            text=json.dumps(ResponseDict(id=request_id, result=None,
+                error=ErrorDict(code=RPCError.WALLET_ERROR,
+                    message=f"Ambiguous account (found {len(accounts)}, expected 1)"))))
+
+    account = accounts[0]
+    # Compatibility: Raises RPC_INVALID_PARAMETER if we were given unlisted named parameters.
+    parameter_values = transform_parameters(request_id, [ "txid", "include_watchonly" ],
+        parameters)
+    # Compatibility: Raises RPC_TYPE_ERROR for non-string (node check allows nulls).
+    txid = parameter_values[0]
+    if parameter_values[0] is not None:
```

Comment:
Please follow the documentation or show it is wrong and correct it. The node does not allow nulls. It explicitly looks up a string raising a RPC_PARSE_ERROR. I have no idea where you got the allows nulls and type error from.

![image](https://user-images.githubusercontent.com/32600614/233928252-78e8eed3-5492-4f01-8333-33e283dc0ad0.png)

![image](https://user-images.githubusercontent.com/32600614/233928120-dc8171ab-7776-49d9-9cf3-42bc09f5365b.png)

![image](https://user-images.githubusercontent.com/32600614/233928181-70fb7f42-3792-4fd6-80f5-a1adbec2c16c.png)

Reply:
I would have copied it from `jsonrpc_signrawtransaction_async` where the same code comment is there for all string parameters (even the first string argument which is required).

But I've removed this comment now. Thanks.

Reply:
Surely you should have removed the if clause and just had the two checks directly?